kyaml/yaml/const.go

```diff
@@ -28,3 +28,35 @@ const (
 	NamespaceField   = "namespace"
 	LabelsField      = "labels"
 )
+
+const (
+	// internalPrefix is the prefix given to internal annotations that are used
+	// internally by the orchestrator
+	internalPrefix string = "internal.config.kubernetes.io/"
+
+	// IndexAnnotation records the index of a specific resource in a file or input stream.
```

Comment:
I was in the impression that these two annotations should not be used changed by KRM function developers. Are we changing this rule? (now they are exposed via   `IndexAnnotation` and `PathAnnotation`

Reply:
Here's the [doc](https://github.com/kubernetes-sigs/kustomize/blob/ca5d691199379db5ef47de539d365e7f664eea24/cmd/config/docs/api-conventions/functions-spec.md#internalconfigkubernetesiopath)
"A function SHOULD NOT modify these annotations."

Reply:
Quoted from the fn spec

> Annotation prefix internal.config.kubernetes.io is reserved for use for internal annotations. In general, a function MUST NOT modify these annotations with the exception of the specific annotations listed below.

A should function **should not** modify the index and path annotations unless it knows what it's doing.